robot-server/robot_server/protocols/analysis_store.py

```diff
@@ -21,18 +23,19 @@ class AnalysisStore:
 
     def __init__(self) -> None:
         """Initialize the AnalysisStore's internal state."""
-        self._analysis_ids_by_protocol: Dict[str, Set[str]] = {}
+        self._analysis_ids_by_protocol: Dict[str, List[str]] = defaultdict(list)
         self._analyses_by_id: Dict[str, ProtocolAnalysis] = {}
 
-    def add_pending(self, protocol_id: str, analysis_id: str) -> List[ProtocolAnalysis]:
+    def add_pending(self, protocol_id: str, analysis_id: str) -> PendingAnalysis:
         """Add a pending analysis to the store."""
-        self._analyses_by_id[analysis_id] = PendingAnalysis.construct(id=analysis_id)
+        ids_for_protocol = self._analysis_ids_by_protocol[protocol_id]
+        assert analysis_id not in ids_for_protocol, "Duplicate analysis ID"
```

Comment:
Do we also want to assert:

```suggestion
        assert analysis_id not in ids_for_protocol, "Duplicate analysis ID"
        assert analysis_id not in self._analyses_by_id, "Duplicate analysis ID"
```